### PR TITLE
return file-polling errors to the caller instead of throwing th

### PR DESCRIPTION
**Pull request by vtjnash**
this allows the user to poll for the creation of a non-existent file

### CODE REVIEW

**Comment by kshyatt** [comment]
Needs a rebase in addition to tests?

**Comment by vtjnash** [comment]
Yes. However, I'm assuming this API change wouldn't be appropriate for the feature freeze?

**Comment by StefanKarpinski** [comment]
How serious is the API change? Does this just add new functionality, or will it break working code?

**Comment by vtjnash** [comment]
See the test breakage for how it could confuse existing code. I'm not sure if there's any existing code, or the exact preferences of those individuals for the behavior of this code.

**Comment by StefanKarpinski** [comment]
Can you just give me a straight answer instead of making me dig through a bunch of logs and then find the location where the thing is called that causes that error?

**Comment by vtjnash** [comment]
updated tests to show how this affects the error case for users of `poll_file`

**Comment by vtjnash** [comment]
OK to merge?

**Comment by StefanKarpinski** [comment]
I think you're in the best position to know, so merge if you think so.

**Comment by vtjnash** [comment]
Right, I like it. The main thing that this changes is that it avoids using errors for control flow, and instead opts for making the error handling an explicit (type-based / type-unstable) feature of the API.

**Comment by StefanKarpinski** [comment]
@JeffBezanson – do you have any thoughts on this kind of API?

**Comment by vtjnash** [comment]
Should I just merge it and see what comes of it?

**Comment by tkelman** [comment]
Does this need documentation updates?

File: base/poll.jl [hunk 1]
```diff
@@ -512,15 +512,20 @@ function watch_file(s::AbstractString, timeout_s::Real=-1)
...
-    poll_file(path::AbstractString, interval_s::Real=5.007, timeout_s::Real=-1) -> (previous::StatStruct, current::StatStruct)
+    poll_file(path::AbstractString, interval_s::Real=5.007, timeout_s::Real=-1) -> (previous::StatStruct, current)
 
 Monitor a file for changes by polling every `interval_s` seconds until a change occurs or
 `timeout_s` seconds have elapsed. The `interval_s` should be a long period; the default is
 5.007 seconds.
 
-Returns a pair of `StatStruct` objects `(previous, current)` when a change is detected.
+Returns a pair of status objects `(previous, current)` when a change is detected.
+The `previous` status is always a `StatStruct`, but it may have all of the field zeroed
```
Comment (tkelman) [inline base/poll.jl]: all of the fields

**Pull request merged**